vm/2.0/modules/ROOT/pages/vm-publish-listen.adoc

```diff
@@ -2,36 +2,102 @@
 :page-aliases: connectors::vm-consume.adoc, connectors::vm/vm-publish-listen.adoc
 
 
-The following example shows how to publish a message that is received by a different flow. To do this, use the `<vm:publish>` and `<vm:listener>` elements that point to the same queue.
+The following example illustrates how to publish a message that is received by a different flow. The first flow takes an incoming HTTP message payload, then the *VM Publish* operation publishes the message to a VM queue. In the second flow, the *VM Listener* source receives the message payload from the previous flow via the same VM queue. To do this, point both operations to the same queue which prevents message loss caused by sending messages to a queue that nobody is listening to.
 
-. Define a flow named `listener` that receives the published message from the subsequent `publish` flow. The `listener` flow contains `<vm:listener>` as a message source and points to `queueName="myQueue"`:
+The following screenshot shows the Anypoint Studio app flow for this example:
 
-[source,xml,linenums]
-----
-<flow name="listener">
-   <vm:listener queueName="myQueue" config-ref="vm"/>
-   <flow-ref name="yourIntegrationLogic"/>
-</flow>
-----
+.VM Connector Publish and Listen to Messages flow
+image::vm-example-publishlisten.png[VM Connector Publish and Listen to Messages flow]
+
+
+== Create the VM Publishing Flow
+
+The VM publishing flow first listens for an incoming HTTP message payload which contains the body attribute of a JSON object, then in a one-way operation the *VM Publish*  publishes the message in a VM queue, and the flow continues.
+
+To create the VM publishing flow: +
+
+. Create a new Mule project in Studio.
+. In the *Mule Palette* view, select the *HTTP Listener* source and drag it into the canvas. +
+The source listens for incoming HTTP messages attributes.
```

Comment:
I think you can just say the source listens for incoming messages... If not, then: 
The source listens for incoming HTTP message attributes.